### PR TITLE
k8s: Add a Go project with container registry for GitOps enabled flow

### PR DESCRIPTION
**Pull request by briancain**

### CODE REVIEW

File: kubernetes/go-gitops/waypoint.hcl [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  default     = ""
+  type        = string
+  description = "password for registry" // DO NOT COMMIT YOUR PASSWORD TO GIT
+}
+
+variable "regcred_secret" {
+  default     = "regcred"
+  type        = string
+  description = "The existing secret name inside Kubernetes for authenticating to the container registry"
+}
```
Comment (catsby) [inline kubernetes/go-gitops/waypoint.hcl:33]: How is this established in the Kube cluster? Should we document it in the README?

File: kubernetes/go-gitops/waypoint.hcl [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  type        = string
+  description = "username for container registry"
+}
+
+// set your registry password through the UI or on the CLI!
+variable "registry_password" {
+  default     = ""
+  type        = string
+  description = "password for registry" // DO NOT COMMIT YOUR PASSWORD TO GIT
+}
```
Comment (catsby) [inline kubernetes/go-gitops/waypoint.hcl:27]:
```suggestion
variable "registry_password" {
  default     = ""
  type        = string
  description = "password for registry" // DO NOT COMMIT YOUR PASSWORD TO GIT
  env = ["REGISTRY_PASSWORD"]
}
```

And maybe add `env = ["REGISTRY_USERNAME"]` as well. Later we could go even further and instruct how to set these envs in a runner profile(?)

**Review by catsby — COMMENTED** [review]
LGTM but I think we need to indicate how to specify the kube secret, specifically it's contents

File: kubernetes/go-gitops/waypoint.hcl (hunk 1 above)
Comment (briancain) [reply kubernetes/go-gitops/waypoint.hcl:33]: Before I set up the secret, I authenticate with docker to my container registry. Then I do something like this, but it's a generic kubernetics secret named `regcred`

```
kubectl create secret generic regcred --from-file=.dockerconfigjson=$HOME/.docker/config.json --type=kubernetes.io/dockerconfigjson
```

Not required I don't think for every workflow, just depends on how they're using container registries!

File: kubernetes/go-gitops/waypoint.hcl (hunk 2 above)
Comment (briancain) [reply kubernetes/go-gitops/waypoint.hcl:27]: Hm, I don't like suggesting people put container registry passwords in their environment. If you wanted to, I think you'd have to set it in a runner profile yeah.

I think the easiest way is to just tell people to set these input variables through the project config in the UI or on the command line if running Waypoint locally (via `waypoint up -var registry_password=blah`)

**Review by catsby — APPROVED** [review]

**Pull request merged**